include/swift/AST/DiagnosticsSema.def

```diff
@@ -3617,6 +3619,10 @@ NOTE(objc_inference_swift3_addnonobjc,none,
 ERROR(objc_for_generic_class,none,
       "generic subclasses of '@objc' classes cannot have an explicit '@objc' "
       "because they are not directly visible from Objective-C", ())
+ERROR(objc_for_resilient_class,none,
+      "classes built with library evolution support cannot have explicit "
+      "'@objc' subclasses because they are not directly "
+      "visible from Objective-C", ())
```

Comment:
As discussed yesterday, this error message doesn't communicate the right thing to the subclass author who finds themself in this situation.

Reply:
We need a diagnostic for the category case too, right?

Reply:
There is one already